pkg/controller/elasticsearch/driver/pvc_expansion_test.go

```diff
@@ -0,0 +1,632 @@
+			err = k8sClient.List(&retrievedSsets)
+			require.NoError(t, err)
+			for i := range tt.wantSsets {
+				comparison.RequireEqual(t, &tt.wantSsets[i], &retrievedSsets.Items[i])
+			}
+
+			var retrievedPods corev1.PodList
+			err = k8sClient.List(&retrievedPods)
+			require.NoError(t, err)
+			for i := range tt.wantPods {
+				comparison.RequireEqual(t, &tt.wantPods[i], &retrievedPods.Items[i])
+			}
+		})
+	}
+}
+
+var (
+	sampleEs = esv1.Elasticsearch{ObjectMeta: metav1.ObjectMeta{Namespace: "ns", Name: "es", UID: "es-uid"}, TypeMeta: metav1.TypeMeta{Kind: "Elasticsearch"}}
+	sset1    = appsv1.StatefulSet{ObjectMeta: metav1.ObjectMeta{Namespace: "ns", Name: "sset1", UID: "sset1-uid"}}
+	pod1     = corev1.Pod{ObjectMeta: metav1.ObjectMeta{Namespace: "ns", Name: "sset1-0", Labels: map[string]string{
+		label.StatefulSetNameLabelName: sset1.Name,
+	}}}
+	pod2 = corev1.Pod{ObjectMeta: metav1.ObjectMeta{Namespace: "ns", Name: "sset1-1", Labels: map[string]string{
+		label.StatefulSetNameLabelName: sset1.Name,
+	}}}
+	pod1WithOwnerRef = *pod1.DeepCopy()
+	pod2WithOwnerRef = *pod2.DeepCopy()
+)
+
+func init() {
+	controllerscheme.SetupScheme()
+	if err := controllerutil.SetOwnerReference(&es, &pod1WithOwnerRef, scheme.Scheme); err != nil {
+		panic(err)
+	}
+	if err := controllerutil.SetOwnerReference(&es, &pod2WithOwnerRef, scheme.Scheme); err != nil {
+		panic(err)
+	}
+}
+
+func Test_updatePodOwners(t *testing.T) {
+	type args struct {
+		k8sClient   k8s.Client
+		es          esv1.Elasticsearch
+		statefulSet appsv1.StatefulSet
+	}
+	tests := []struct {
+		name     string
+		args     args
+		wantPods []corev1.Pod
+	}{
+		{
+			name: "happy path: set an owner ref to the ES resource on all Pods for that StatefulSet",
+			args: args{
+				k8sClient:   k8s.WrappedFakeClient(&pod1, &pod2),
+				es:          sampleEs,
+				statefulSet: sset1,
+			},
+			wantPods: []corev1.Pod{pod1WithOwnerRef, pod2WithOwnerRef},
+		},
+		{
+			name: "owner ref already set: the function is idempotent",
+			args: args{
+				k8sClient:   k8s.WrappedFakeClient(&pod1WithOwnerRef, &pod2WithOwnerRef),
+				es:          sampleEs,
+				statefulSet: sset1,
+			},
+			wantPods: []corev1.Pod{pod1WithOwnerRef, pod2WithOwnerRef},
+		},
+		{
+			name: "one owner ref already set, one missing",
+			args: args{
+				k8sClient:   k8s.WrappedFakeClient(&pod1WithOwnerRef, &pod2),
+				es:          sampleEs,
+				statefulSet: sset1,
+			},
+			wantPods: []corev1.Pod{pod1WithOwnerRef, pod2WithOwnerRef},
+		},
+		{
+			name: "no Pods: nothing to do",
+			args: args{
+				k8sClient:   k8s.WrappedFakeClient(),
+				es:          sampleEs,
+				statefulSet: sset1,
+			},
+			wantPods: []corev1.Pod{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := updatePodOwners(tt.args.k8sClient, tt.args.es, tt.args.statefulSet)
+			require.NoError(t, err)
+
+			var retrievedPods corev1.PodList
+			err = tt.args.k8sClient.List(&retrievedPods)
+			require.NoError(t, err)
```

Comment:
`wantPods` is never used, I think we want to compare it with `retrievedPods` ?

Reply:
🤦 👍